components/admin_console/schema_admin_settings.test.jsx

```diff
@@ -317,4 +318,60 @@ describe('components/admin_console/SchemaAdminSettings', () => {
             />,
         )).toEqual(true);
     });
+
+    test('should not try to validate when a setting does not contain a key', () => {
+        const mockValidate = jest.fn(() => {
+            return new ValidationResult(true, '', '');
+        });
+
+        const localSchema = {...schema};
+        localSchema.settings = [
+            {
+
+                // won't validate because no key
+                label: 'a banner',
+                type: 'banner',
+            },
+        ];
+        const props = {
+            config,
+            environmentConfig,
+            schema: localSchema,
+            updateConfig: jest.fn(),
+        };
+
+        const wrapper = shallow(<SchemaAdminSettings {...props}/>);
+
+        expect(wrapper.instance().canSave()).toBe(true);
+        expect(mockValidate).not.toHaveBeenCalled();
```

Comment:
The `setting` should have `validate: mockValidate` for this assert, right?

Reply:
Yeah that would make sense to add it!